### ISSUE TITLE
Slack file.upload API with remote renderer works but image not uploaded to Slack notification

### ISSUE THREAD

**Issue by 1am** [grafana/grafana#23518]
**What happened**:

Alert over Slack channel is triggered correctly but does not include an image. I'm trying to use the file.upload Slack API and have the rendering done by https://grafana.com/docs/grafana/latest/administration/image_rendering/#remote-rendering-service

The logs from Grafana look like it would generate the image correctly, execute the Slack API request ok but the notification only shows text without image.

```
grafana_1     | t=2020-04-11T22:44:01+0000 lvl=info msg="New state change" logger=alerting.resultHandler ruleId=1 newState=ok prev state=alerting
grafana_1     | t=2020-04-11T22:44:01+0000 lvl=info msg=Rendering logger=rendering renderer=http path="d-solo/Qu3owYCZl/telegraf-tests?orgId=1&panelId=8"
grafana_1     | t=2020-04-11T22:44:01+0000 lvl=dbug msg="calling remote rendering service" logger=rendering renderer=http url="http://renderer:8081/render?domain=grafana&encoding=&height=500&renderKey=9v9YokKMeCXjprAm3K9iy2uJF949y1ZX&timeout=15&timezone=&url=http%3A%2F%2Fgrafana%3A3000%2Fd-solo%2FQu3owYCZl%2Ftelegraf-tests%3ForgId%3D1%26panelId%3D8%26render%3D1&width=1000"
renderer_1    | {"url":"http://grafana:3000/d-solo/Qu3owYCZl/telegraf-tests?orgId=1&panelId=8&render=1","level":"debug","message":"Render request received"}
renderer_1    | {"file":"/tmp/5e09f2ff.png","level":"debug","message":"Deleting temporary file"}
renderer_1    | {"url":"http://grafana:3000/d-solo/Qu3owYCZl/telegraf-tests?orgId=1&panelId=8&render=1","level":"debug","message":"Connection closed"}
renderer_1    | {"message":"::ffff:172.27.0.2 - - [11/Apr/2020:22:44:04 +0000] \"GET /render?domain=grafana&encoding=&height=500&renderKey=9v9YokKMeCXjprAm3K9iy2uJF949y1ZX&timeout=15&timezone=&url=http%3A%2F%2Fgrafana%3A3000%2Fd-solo%2FQu3owYCZl%2Ftelegraf-tests%3ForgId%3D1%26panelId%3D8%26render%3D1&width=1000 HTTP/1.1\" 200 22049 \"-\" \"Grafana/6.7.2\"\n","level":"debug"}
grafana_1     | t=2020-04-11T22:44:04+0000 lvl=info msg="Executing slack notification" logger=alerting.notifier.slack ruleId=1 notification=slack-test-channel-1
grafana_1     | t=2020-04-11T22:44:04+0000 lvl=info msg="Uploading to slack via file.upload API" logger=alerting.notifier.slack
```

It seems to be very similar to what @ZealousMacwan mentioned in  https://github.com/grafana/grafana/issues/20885 and also from reading https://github.com/grafana/grafana/issues/21018#issuecomment-575579432 it seems to be closed with 6.6.0 but I'm experiencing this with 6.7.2. Also reading https://github.com/grafana/grafana/issues/11059 I have set 'Recipient' field to notification channel name.

**What you expected to happen**:

The alert message on Slack would include a snapshot of the alert generating chart.

**How to reproduce it (as minimally and precisely as possible)**:

- Ubuntu 18 hosted docker-compose setup with Grafana 6.7.2 + Grafana image renderer 1.0.12 as described in https://github.com/grafana/grafana-image-renderer/blob/master/docs/remote_rendering_using_docker.md#docker-compose-example
- Validate that the instructions for manual creation of snapshot via "Panel context menu -> Share -> Direct link rendered image" work ok
- Set up an alert with Slack notification channel and press "Send test" - no image here though not sure there should be any
- Create an alert on any panel and trigger it. Slack notification should be sent but it will not contain any image 
![Screen Shot 2020-04-12 at 00 56 53](https://user-images.githubusercontent.com/959323/79056606-9232f200-7c58-11ea-8711-d30994196713.png)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.7.2
- Data source type & version: Influxdb 1.7.10
- OS Grafana is installed on: Ubuntu 18.04.3 LTS
- User OS & Browser: Ubuntu 18.04.3 LTS Chromium
- Grafana plugins: None
- Others: Docker compose setup like in https://github.com/grafana/grafana-image-renderer/blob/master/docs/remote_rendering_using_docker.md#docker-compose-example

**Comment by hugohaggmark** [comment]
Thanks for reporting this @1am, but I just setup the same thing using the docker-compose file mentioned for the first time and everything works fine. The test slack test notification should send an image:
![image](https://user-images.githubusercontent.com/562238/79189913-3ec0cf80-7e23-11ea-9776-d543cd3c2671.png)

Maybe there's something configured wrong, here is my slack notification configs:
![image](https://user-images.githubusercontent.com/562238/79190064-9f500c80-7e23-11ea-8bcd-1ebc71fe089f.png)

**Comment by 1am** [comment]
Hello @hugohaggmark Thank you for your response and help. I had the same settings in Grafana but the image wasn't getting through. I started investigating on Slack bot side and it turns our that I just had to explicitly grant file.write permissions to it which I've missed initially while reading the docs.

![Screen Shot 2020-04-14 at 08 36 22](https://user-images.githubusercontent.com/959323/79193672-661b9a80-7e2b-11ea-9da8-71f5ed105deb.png)
